src/store/messagesStore.js

```diff
@@ -1256,30 +1257,49 @@ const actions = {
 	},
 
 	/**
-	 * Posts a simple text message to a room
+	 * Forwards message to a conversation
 	 *
 	 * @param {object} context default store context;
 	 * will be forwarded;
 	 * @param {object} data the wrapping object;
+	 * @param {object} data.targetToken the conversation token to where the message will be forwarded;
 	 * @param {object} data.messageToBeForwarded the message object;
 	 */
-	async forwardMessage(context, { messageToBeForwarded }) {
-		const response = await postNewMessage(messageToBeForwarded, { silent: false })
-		return response
-	},
+	async forwardMessage(context, { targetToken, messageToBeForwarded }) {
+		const message = cloneDeep(messageToBeForwarded)
+		// Overwrite with the target conversation token
+		message.token = targetToken
+
+		if (message.parent) {
+			delete message.parent
+		}
+
+		if (message.message === '{object}' && message.messageParameters.object) {
+			const richObject = message.messageParameters.object
+			await postRichObjectToConversation(
+				targetToken,
+				{
+					objectId: richObject.id,
+					objectType: richObject.type,
+					metaData: JSON.stringify(richObject),
+					referenceId: '',
+				},
+			)
+			return
+		}
 
-	/**
-	 * Posts a simple text message to a room
-	 *
-	 * @param {object} context default store context;
-	 * will be forwarded;
-	 * @param {object} data the wrapping object;
-	 * @param {string} data.token token of the target conversation
-	 * @param {object} data.richObject the rich object;
-	 */
-	async forwardRichObject(context, { token, richObject }) {
-		const response = await postRichObjectToConversation(token, richObject)
+		// If there are mentions in the message to be forwarded, replace them in the message
+		// text.
+		for (const key in message.messageParameters) {
+			if (key.startsWith('mention')) {
+				const mention = message.messageParameters[key]
+				message.message = message.message.replace(`{${key}}`, `@'${mention.name}'`)
+			}
+		}
+
+		const response = await postNewMessage(message, { silent: false })
```

Comment:
Maybe it should be silent, if posting to yourself? And counter ommited also?

Reply:
Do you mean unread counter? If yes, it won't have it because all messages are coming from the current user.